src/components/Carousel.js

```diff
@@ -339,15 +343,29 @@ class Carousel extends Component {
 
     onSwipeStart = () => {
         this.setState({
-            swiping: true
+            swiping: true,
         });
+        if (this.props.infiniteLoop) {
+            this.setState({
+                swipeLoop: true,
+            });
+        }
         this.clearAutoPlay();
     }
 
     onSwipeEnd = () => {
         this.setState({
             swiping: false
         });
+        if (this.props.infiniteLoop) {
+            requestAnimationFrame(() => {
```

Comment:
This seems too fragile IMO. Shouldn't you have been using a callback in the setState call instead?

Reply:
This is for the same reason to use nested `rAF` in `getPosition`. We need two animations, thus in two frames. `setState` callback may schedule state change in the same frame (as it uses micro task to do scheduling), so style changes within the same frame, then it leads to unexpected animation.

Reply:
Answered there...